cdap-watchdog/src/main/java/co/cask/cdap/logging/pipeline/kafka/KafkaLogProcessorPipeline.java

```diff
@@ -368,6 +369,20 @@ private int appendEvents(long currentTimeMillis, boolean forced) {
       eventsAppended++;
     }
 
+    // For each partition, if there is no more event in the event queue, update the checkpoint nextOffset
+    for (Int2LongMap.Entry entry : offsets.int2LongEntrySet()) {
+      int partition = entry.getIntKey();
+      if (eventQueue.isEmpty(partition)) {
+        MutableCheckpoint checkpoint = checkpoints.get(partition);
+        long offset = entry.getLongValue();
+        // If the process offset is larger than the offset in the checkpoint and the queue is empty for that partition,
+        // it means everything before the process offset must had been written to the appender.
+        if (checkpoint != null && offset > checkpoint.getNextOffset()) {
```

Comment:
when will this case happen ? as we update the checkpoint whenever we dequeue from the eventQueue.